submissions/webdevtoday/friends-app/js/app.js

```diff
@@ -0,0 +1,95 @@
+import { getUsers } from './components/RandomUserAPI.js';
+import { normalizeUsers } from './components/normalizeUsers.js';
+import { makeCard } from './components/makeCard.js';
+import { oops } from './components/oops.js';
+import { makeForm } from './components/makeForm.js';
+import { sortAndFiltersFunctions } from './components/sortAndFiltersFunctions.js';
+
+let users = [];
+const sortAndFiltersState = {
+    // filters:
+    age: null,
+    name: null,
+    gender: null,
+
+    // sorts:
+    sort: null,
+};
+
+const sidebar = document.getElementsByClassName('sidebar')[0];
+const sidebarOpener = document.getElementsByClassName('sidebar-opener')[0];
+const sidebar__container = document.getElementsByClassName('sidebar__container')[0];
+const cards = document.getElementsByClassName('cards')[0];
+
+function addListenerToAsideOpener() {
+    sidebarOpener.addEventListener('click', () => {
+        sidebarOpener.classList.toggle('wrapper__sidebar-opener--active');
+        sidebar.classList.toggle('wrapper__sidebar--active');
+    });
+}
+
+function addListenerToResetButton() {
+    document.querySelector('[type=reset]').addEventListener('click', () => {
+        sortAndFiltersState.age = null;
+        sortAndFiltersState.name = null;
+        sortAndFiltersState.gender = null;
+        sortAndFiltersState.sort = null;
```

Comment:
what if you have 50 such fields, will you manually iterate over them?

Reply:
I've fixed it.